db.go

```diff
@@ -110,6 +110,12 @@ type DB struct {
 	// of truncate() and fsync() when growing the data file.
 	AllocSize int
 
+	// MaxSize is the maximum amount of space allowed for the data file.
+	// If a caller's attempt to add data results in the need to grow
+	// the data file, an error will be returned and the data file will not grow.
+	// 0 means no maximum.
+	MaxSize int
```

Comment:
This feature looks good to me.

I have one question: If the original file size is higher than MaxSize, do we allow to open db or just return error?

cc @ahrtr

Reply:
I've added this test now